diff --git a/node_modules/preact-cli-plugin-netlify/index.js b/node_modules/preact-cli-plugin-netlify/index.js
index 7187105..2681077 100644
--- a/node_modules/preact-cli-plugin-netlify/index.js
+++ b/node_modules/preact-cli-plugin-netlify/index.js
@@ -4,8 +4,9 @@
  */
 
 class NetlifyServerPushPlugin {
-  constructor({ redirects }) {
+  constructor({ redirects, brotli = false }) {
     this.redirects = [];
+    this.brotli = brotli;
     if (redirects !== undefined) {
       if (Array.isArray(redirects)) {
         this.redirects = redirects;
@@ -39,6 +40,10 @@ class NetlifyServerPushPlugin {
       let headers =
         '/*\n\tCache-Control: public, max-age=3600, no-cache\n\tAccess-Control-Max-Age: 600\n/sw.js\n\tCache-Control: private, no-cache\n/*.chunk.*.js\n\tCache-Control: public, max-age=31536000';
 
+      if (this.brotli) {
+        headers += '\n/*.br\n\tcontent-encoding: br';
+      }
+
       const redirects = `${this.redirects.join('\n')}\n/* /index.html 200`;
 
       routes.forEach(filename => {
@@ -81,6 +86,17 @@ module.exports = function(config, options = {}) {
       'You need to pass the webpack config to preact-cli-plugin-netlify!',
     );
   }
+
+  const swBuilder = getPluginsByName(config, 'SWBuilderPlugin');
+  if (
+    swBuilder &&
+    swBuilder.length > 0 &&
+    swBuilder[0].plugin &&
+    swBuilder[0].plugin.brotli_
+  ) {
+    options.brotli = true;
+  }
+
   config.plugins.push(new NetlifyServerPushPlugin(options));
   const plugins = config.plugins;
   for (let pluginIndex = 0; pluginIndex < plugins.length; pluginIndex++) {
@@ -97,3 +113,13 @@ module.exports = function(config, options = {}) {
     }
   }
 };
+
+function getPluginsByName(config, name) {
+  return getPlugins(config).filter(
+    w => w.plugin && w.plugin.constructor && w.plugin.constructor.name === name,
+  );
+}
+
+function getPlugins(config) {
+  return (config.plugins || []).map((plugin, index) => ({ index, plugin }));
+}
